apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/DcGeoRDALoadJob.java

```diff
@@ -0,0 +1,149 @@
+  public static final String SCAN_INTERVAL_PROPERTY = "DCGeoRDALoadIntervalSeconds";
+  public static final String RUN_TIME_PROPERTY = "DCGeoRDALoadRunSeconds";
+  public static final String MAX_RECORDS_PROPERTY = "DCGeoRDALoadMaxRecords";
+  public static final int SCAN_INTERVAL_DEFAULT = 300;
+  public static final int RUN_TIME_DEFAULT = 300;
+  public static final int MAX_RECORDS_DEFAULT = Integer.MAX_VALUE;
+  public static final String BATCH_SIZE_PROPERTY = "DCGeoBatchSize";
+  public static final int BATCH_SIZE_DEFAULT = 1;
+  public static final String CALLS_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "calls");
+  public static final String FAILURES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "failures");
+  public static final String SUCCESSES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "successes");
+  public static final String PROCESSED_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "processed");
+
+  private final Config config;
+  private final Callable<RDASource<PreAdjudicatedClaim>> sourceFactory;
+  private final Callable<RDASink<PreAdjudicatedClaim>> sinkFactory;
+  private final Meter callsMeter;
+  private final Meter failuresMeter;
+  private final Meter successesMeter;
+  private final Meter processedMeter;
+
+  public DcGeoRDALoadJob(
+      Config config,
+      Callable<RDASource<PreAdjudicatedClaim>> sourceFactory,
+      Callable<RDASink<PreAdjudicatedClaim>> sinkFactory,
+      MetricRegistry appMetrics) {
+    this.config = config;
+    this.sourceFactory = sourceFactory;
+    this.sinkFactory = sinkFactory;
+    callsMeter = appMetrics.meter(CALLS_METER_NAME);
+    failuresMeter = appMetrics.meter(FAILURES_METER_NAME);
+    successesMeter = appMetrics.meter(SUCCESSES_METER_NAME);
+    processedMeter = appMetrics.meter(PROCESSED_METER_NAME);
+  }
+
+  /**
+   * Factory method to construct a new job instance using standard parameters.
+   *
+   * @param appMetrics MetricRegistry used to track operational metrics
+   * @return a DcGeoRDALoadJob instance suitable for use by PipelineManager.
+   */
+  public static DcGeoRDALoadJob newDcGeoRDALoadJob(MetricRegistry appMetrics) {
+    return new DcGeoRDALoadJob(
+        new Config(),
+        () -> new SkeletonRDASource(appMetrics),
+        () -> new SkeletonRDASink(appMetrics),
+        appMetrics);
+  }
+
+  public Duration getScanInterval() {
+    return config.getScanInterval();
+  }
+
+  @Override
+  public PipelineJobOutcome call() throws Exception {
+    final long startMillis = System.currentTimeMillis();
+    int processedCount = 0;
+    Exception error = null;
+    try {
+      callsMeter.mark();
+      try (RDASource<PreAdjudicatedClaim> source = sourceFactory.call();
+          RDASink<PreAdjudicatedClaim> sink = sinkFactory.call()) {
+        processedCount =
+            source.retrieveAndProcessObjects(
```

Comment:
if the upcoming changes to the PipelineManager supports threading and allows for us to leave the connection open indefinitely; our metrics wouldn't move until a shutdown.  No need to address now, but we might need to rethink how to manage metrics if we want to leave the connection open

Reply:
That's an excellent point.  There are a few other points to consider though:

- the RDASource/RDASink implementation objects will define their own internal metrics
- the processedMeter increments in tandem with the callsMeter which has the same behavior you mentioned.
- the limits on run time, idle time, and number of objects streamed in our calls to RDA will tend to cause these calls to last a finite amount of time (assuming we configure them)

The actual metrics we record definitely needs some more consideration/discussion going forward.

Reply:
Yeah lets plan to talk about KPIs in Sprint 2.2, no sense in overthinking this before we have a clear picture of what we want to track.